Add mark-yank

mkleehammer opened the pull request:
### Brief summary of what the package does

This simple Emacs minor mode provides the command mark-yank that marks the last yank / paste in the current buffer.

When the mode is enabled, it advises the yank command and records the location. When the mark-yank command is invoked, it sets the mark and point to the recorded location and activates the region.

### Direct link to the package repository

https://github.com/mkleehammer/mark-yank

### Your association with the package

Author & maintainer

### Relevant communications with the upstream package maintainer

**None needed**

### Checklist

<!-- Please confirm by replacing `[]` with `[x]`: -->

- [x] The package is released under a [GPL-Compatible Free Software License](https://www.gnu.org/licenses/license-list.en.html#GPLCompatibleLicenses)
- [x] I've read [CONTRIBUTING.org](https://github.com/melpa/melpa/blob/master/CONTRIBUTING.org)
- [x] I've used the latest version of [package-lint](https://github.com/purcell/package-lint) to check for packaging issues, and addressed its feedback
- [x] My elisp byte-compiles cleanly
- [x] I've used `M-x checkdoc` to check the package's documentation strings
- [x] I've built and installed the package using the instructions in [CONTRIBUTING.org](https://github.com/melpa/melpa/blob/master/CONTRIBUTING.org)

<!-- After submitting, please fix any problems the CI reports. -->

alphapapa commented:
It would be helpful to compare this with the package `volatile-highlights`.

jcs090218 commented:
I'm helping MELPA maintainers a little bit here.

- No byte-compile error 🎉 

`mark-yank.el` with checkdoc (0.6.2)

```
mark-yank.el:90: The footer should be: (provide 'mark-yank)\n;;; mark-yank.el ends here
```

`mark-yank.el` with package-lint (20230905.629)

```
No issues found.
```

Package and license:

- MIT

riscy commented:
Thanks for this.  I noticed you've got other PR(s) already open #8791.  To split my review time evenly between authors, I try to do one PR at a time per author, but I promise I'll get to this when we resolve earlier PRs.

mkleehammer commented:
I completely understand.  Thanks for taking the time to do this - package.el and MELPA breathed new life into the Emacs ecosystem.

I've fixed the provide keyword.  I renamed the buffer to remove "-mode" and forgot to update the provide line.  I apparently forgot to rerun checkdoc afterwards.

I've also rewrapped everything at 80 characters.  (I will make that fill column change in my lisp hook function.)

riscy commented:
Thanks for accommodating - I think this is still tripping up package-lint.  The reason is your first line is:
```
;;; mark-yank-mode.el --- Set region to the last yank  -*- lexical-binding: t -*-
```
but the file's name is mark-yank.el, not mark-yank-mode.el, and package-lint is looking for a match which it's not finding.

Some other feedback:
- `(message "mark-yank-mode %s" (if mark-yank-mode "enabled" "disabled"))` is unnecessary - you will find that a similar message is already echoed automatically via `define-minor-mode`'s conventions
- You don't need a `progn` for the "else" part of an `if` (in `#'mark-yank`)
- In `mark-yank--advise-after` are you getting any benefit at all from checking `(markerp mark-yank--last-mark)`?  (Why not always just create a new pair of markers?  I may be missing something.)  If not, I think you might eliminate a potential codepath for bugs by just doing the same thing every time
- `mark-yank--last-mark` and `mark-yank--last-point` are both buffer-local variables (which makes sense to me) but `mark-yank-mode` is a global minor mode -- why not also have it be a local minor mode?  For instance I may want this behavior in my editing buffers, but not in my comint-mode buffers

But altogether I think this seems like a novel new editing command.

mkleehammer commented:
Thanks for the feedback.  I've fixed the issue and made it a minor & global mode.

I had reused the existing markers because I was under the impression they were very expensive.  Maybe that just referred to them being expensive to maintain once created.  (Which is why I originally used ints for the position, but I inevetibly move stuff around and then want to mark the pasted text.  Usually lots of SQL I've worked on in a dedicated SQL buffer and now need to tweak and indent.)

riscy commented:
> I was under the impression they were very expensive.

This may well be true, so that's not a blocker.  I'd be curious to know what you find through any research/benchmarking -- it would fill in a knowledge gap of mine.
